### ISSUE TITLE
NavigationDuplicated error during login 

### ISSUE THREAD

**Issue by sonokoh** [Source-Digital/marketplace-admin#23]
**Describe the bug**
When I access to login page and try to login, I'm getting NavigationDuplicated error.

**To Reproduce**
1. Go to login page
2. Confirm localStorage's `pathToLoadAfterLogin` value is set to `"login"` 
3. Login with valid credential
4. Login successful message is displayed but screen is not navigated to dashboard
5. Console displays the error

**Expected behavior**
After the login user should be navigated to the dashboard

**Screenshots**
<img width="1796" alt="Screen Shot 2020-12-24 at 9 41 37 AM" src="https://user-images.githubusercontent.com/74166775/103095125-7c3e4500-45cd-11eb-9959-ffaad3b84476.png">

**Additional Comments**
I think we can make implementation much simpler -- 

First we get rid of loading page. We can just put some loading component in root component such as App.vue that renders loading animation when application is still loading. This will get rid of the loading state and the logic will be much simpler.

Now, the only time we need redirecting is when token auto-login fails. There should be a code that redirects user to login screen on such login failure. Let's capture the target navigation path user wanted to go there. If user logs in successfully from login page, we can navigate them to their desired destination using the captured path. 
Right now we are capturing the paths on all navigation which I believe is unnecessary.

For storing the target path `localStorage` works but it is tad cumbersome that we need to clean up the property in appropriate timings (because localStorage data persists after page refresh). How about we use `queryParam` instead? When we redirect user to login page, we append `redirect` query to the url. If login page sees the param on login success, the page navigates user to the page rather than home.

**Comment by sonokoh** [comment]
@onexdata @cjshay lmk what you think :)

**Comment by cjshay** [comment]
I made a quick fix and PRed for this, but I agree that the workflow could be cleaner without the loading route!

**Comment by cjshay** [comment]
Agreed that getting rid of the loading page would streamline the application flow, and make solving the redirect problem easier. There is one part of the linkage I haven't had time to get my head around yet, though. Right now, the loading page is shown when the store indicates that the system is not loaded (in router/index.js the condition is if (!store.state.system.loaded)), but it is not clear to me where that flag is getting toggled or what it means for the rest of the application. If that step is not important, then we can cut about half of the route guards.

Agreed on query params being a cleaner way of solving this problem. Part of the motivation for the use of local storage was that holding on to the query params was a bit tricky through the multiple redirects connected to loading.

**Comment by sonokoh** [comment]
So I read the code and it seems we are loading vuex state from experience engine node modules. There's 'state' module which has "loaded" flag. So when `this.$store.commit('loaded', true)` is called in App.vue, the state is toggled.
I believe we can just refer to this flag from App.vue component, show loading indicator while the flag is false and hide everything else.

Yeah I totally understand that. I'll merge your quick patch PR (ty!) but will keep this issue open so that we can discuss further for better design.
